ow_plexil/src/plexil-adapter/LanderAdapter.h

```diff
@@ -40,12 +41,44 @@ class LanderAdapter : public PlexilAdapter
 
  protected:
   template<typename T>
-  auto lookupHandler (const T& val)
+  auto lookupHandler_constant (const T& val)
   {
     return [=] (const PLEXIL::State&, PLEXIL::LookupReceiver* r) {
              r->update(PLEXIL::Value(val));
            };
   }
+
+  template<typename T, typename Class, typename Base>
+  auto lookupHandler_function0 (const Class& instance,
+                                T(Base::*method)() const)
+  {
+    return [&instance, method] (const PLEXIL::State&,
+                                PLEXIL::LookupReceiver* r) {
+             r->update(PLEXIL::Value((instance.*method)()));
+           };
+  }
+
+  template<typename T, typename Class, typename Base>
+  auto lookupHandler_function1 (const Class& instance,
+                                T(Base::*method)(const std::string&) const)
+  {
+    return [&instance, method] (const PLEXIL::State& s,
+                                PLEXIL::LookupReceiver* r) {
+             std::string arg;
+             s.parameters()[0].getValue(arg);
+             r->update(PLEXIL::Value((instance.*method)(arg)));
+           };
+  }
+
+  /* For future reference: an attempt using std::bind() that doesn't build:
+  template <typename T, typename Callable>
+  auto lookupHandler_callable (Callable c)
```

Comment:
No suggesting it for this PR, but I think you could make this work by replacing the `Callable c` arg with an `std::function` that has a templated return value.  Unsure if there is a nice way to handle a string argument lookup with this method.

In any case, I think what  we have here is fine.

Reply:
I'll leave this for future improvement, along with using the new template functions in LanderAdapter.cpp, which will require some more abstraction because it supports lookups with an integer parameter.